examples/grpc-kotlin/src/main/kotlin/example/armeria/grpc/kotlin/HelloServiceImpl.kt

```diff
@@ -3,52 +3,50 @@ package example.armeria.grpc.kotlin
 import com.linecorp.armeria.server.ServiceRequestContext
 import example.armeria.grpc.kotlin.Hello.HelloReply
 import example.armeria.grpc.kotlin.Hello.HelloRequest
-import io.grpc.stub.StreamObserver
-import java.time.Duration
-import java.util.concurrent.TimeUnit
-import reactor.core.publisher.Flux
-import reactor.core.scheduler.Schedulers
-
-class HelloServiceImpl : HelloServiceGrpc.HelloServiceImplBase() {
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.toList
+import kotlinx.coroutines.withContext
+
+/**
+ * Note that if you want to run [HelloServiceImpl] in an event loop and
+ * get a current [ServiceRequestContext],
+ * you should initialize [HelloServiceImpl] with [Dispatchers.Unconfined].
```

Comment:
coroutines running on event loop will be awesome. Does `Dispatchers.Unconfined` allow that to happen? I guess callbacks will return on any arbitrary thread. If that's the case, we should investigate more and probably need to implement `Dispatchers.ArmeriaContext`. Not sure, but maybe it just means creating an implementation of the interface that delegates all methods to `RequestContext.contextAwareExecutor().asCoroutineDispatcher()`

Reply:
For reference, in our non-grpc example, we use this `GlobalScope` pattern and can confirm the context / executor everywhere.

https://github.com/line/armeria/blob/master/examples/context-propagation/kotlin/src/main/kotlin/example/armeria/contextpropagation/kotlin/MainService.kt#L43

Not sure how that corresponds to this constructor though.

Reply:
Thanks for the advice! Let me take a look. 😀

Reply:
> Does Dispatchers.Unconfined allow that to happen? 

`Dispatchers.Unconfined` run coroutines on the current thread. I thought if this method is called by Armeria gRPC service, this will run an event loop.

Reply:
If it works that's great :) Main concern is doc says not "mandating any threading policy" so it sounds like it'd be random.

https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/-dispatchers/-unconfined.html